pkg/plugin/comment_service.go

```diff
@@ -0,0 +1,103 @@
+package plugin
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/arquillian/ike-prow-plugins/pkg/scm"
+	"github.com/arquillian/ike-prow-plugins/pkg/utils"
+	"github.com/google/go-github/github"
+	"github.com/sirupsen/logrus"
+)
+
+// IkePluginsTitle is a constant containing "Ike Plugins" title with markdown formatting
+const IkePluginsTitle = "### Ike Plugins"
+
+// CommentService is a struct managing plugin comments
+type CommentService struct {
+	client         *github.Client
+	log            *logrus.Entry
+	issue          RepositoryIssue
+	commentContext CommentContext
+}
+
+// RepositoryIssue holds owner name, repository name and an issue number
+type RepositoryIssue struct {
+	Owner    string
+	RepoName string
+	Number   int
+}
+
+// CommentContext holds a plugin name and a assignee to be mentioned in the comment
+type CommentContext struct {
+	PluginName string
+	Assignee   string
+}
+
+// NewCommentService creates an instance of GitHub CommentService for the given CommentContext
+func NewCommentService(client *github.Client, log *logrus.Entry, change scm.RepositoryChange, issueOrPrNumber int, commentContext CommentContext) *CommentService {
+	return &CommentService{
+		client: client,
+		log:    log,
+		issue: RepositoryIssue{
+			Owner:    change.Owner,
+			RepoName: change.RepoName,
+			Number:   issueOrPrNumber,
+		},
+		commentContext: commentContext,
+	}
+}
+
+// PluginComment checks all present comments in the issue/pull-request. If no comment with IkePluginsTitle is found, then
+// it adds a new comment with IkePluginsTitle, assignee, the plugin tittle and the given commentMsg. If a comment with
+// IkePluginsTitle is found but missing the particular plugin title, then it modifies the comment by adding plugin title
+// with the given commentMsg. If everything is present already, then it does nothing.
+func (s *CommentService) PluginComment(commentMsg string) error {
```

Comment:
I am not sure if that will work considering all the plugins are notified by the `hook` service at the moment particular event arrives. This might mean we can have a race condition and end up with multiple comments at the same time.

Maybe, for the time being, let's make one comment per plugin to make them independent? We will think about overall comment with references to all active plugins later on. Especially considering the fact that Prow has special `help` website where it lists all plugins (see what is deployed to your openshift cluster)

Reply:
I'm aware of the problem. I have an idea how to solve it, but didn't want to try to implement it here. I thought that this could be fine as the first step...

Reply:
comment-per-plugin 797b9b5